workflow/controller/controller.go

```diff
@@ -85,6 +85,7 @@ type WorkflowController struct {
 	wfArchive             sqldb.WorkflowArchive
 	metrics               metrics.Metrics
 	eventRecorder         record.EventRecorder
+	archiveLabelSelector  labels.Selector
```

Comment:
I don't think we should have this field. We should use the config map as the source of truth instead of using two separate fields. They could get out of sync between updates and produce bugs that are difficult to trace.

Reply:
This field will be always syncd whenever `configmap` updated. it is following same NullObject pattern.